home: Replace /try-zulip with a modal for spectators.

prakhar1144 opened the pull request:
[#backend > replace /try-zulip with modal #34181](https://chat.zulip.org/#narrow/channel/3-backend/topic/replace.20.2Ftry-zulip.20with.20modal.20.2334181/with/2140826)

[#design > Replace /try-zulip with a modal #34181](https://chat.zulip.org/#narrow/channel/101-design/topic/Replace.20.2Ftry-zulip.20with.20a.20modal.20.2334181/with/2136940)

PR to replace [zulip.com/try-zulip](http://zulip.com/try-zulip) with a modal as mentioned in [#34181](https://github.com/zulip/zulip/pull/34181)

<!-- Describe your pull request here.-->

Fixes: #34181 

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

<img width="499" alt="Screenshot 2025-04-02 at 1 18 59 AM" src="https://github.com/user-attachments/assets/7fe0f08c-bd8e-49a1-853c-329386a99df4" />


<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
I might try to trim off a few words, but looks good in general! I think this will feel much smoother than the landing page.

timabbott commented on zerver/lib/home.py:
```diff
@@ -132,12 +133,17 @@ def build_page_params_for_home_page_load(
+        split_url = urlsplit(request.build_absolute_uri())
+        show_try_zulip_modal = (
+            settings.DEVELOPMENT or split_url.hostname == "chat.zulip.org"
+        ) and split_url.query == "show_try_zulip_modal"
```
I'm not sure we need a conditional on DEVELOPMENT or hostname here; we don't expect there to be a reason anyone would pass `?show_try_zulip_modal` on other servers.

timabbott replied:
But it'd be pretty confusing if shown on a random other server, and the check is harmless, so maybe better to have it than not.

timabbott commented on web/templates/try_zulip_modal.hbs:
```diff
@@ -0,0 +1,29 @@
+            {{#*inline "z-design-channel"}}<a href="/#narrow/channel/101-design">{{> @partial-block}}</a>{{/inline}}
+            {{#*inline "z-issues-channel"}}<a href="/#narrow/channel/9-issues">{{> @partial-block}}</a>{{/inline}}
+            {{/tr}}
+        </li>
```
These are channel feed links, I think? We should think about whether we want them to work that way or have JS code to take you to the first topic.

alya replied:
Let's just make these bold-formatted rather than being links; I don't think we need to link them here.

prakhar1144 replied:
Made it bold. See updated screenshot in the PR description.

prakhar1144 commented on web/src/ui_init.js:
```diff
@@ -719,8 +721,33 @@ export function initialize_everything(state_data) {
+        url.searchParams.delete("show_try_zulip_modal");
+        window.history.replaceState(window.history.state, "", url.toString());
+    }
+
```
Made this change in the recent push:
```diff
        focus_submit_on_open: true,
        close_on_submit: true,
-       on_show() {
-         // Remove '?show_try_zulip_modal'
-         const url = new URL(window.location.href);
-         url.searchParams.delete("show_try_zulip_modal");
-         window.history.replaceState(window.history.state, "", url.toString());
-       }
    });
}

$(() => {
+   // Remove '?show_try_zulip_modal', if present.
+   const url = new URL(window.location.href);
+   if (url.searchParams.has("show_try_zulip_modal")) {
+       url.searchParams.delete("show_try_zulip_modal");
+       window.history.replaceState(window.history.state, "", url.toString());
+   }
```

prakhar1144 commented:
Updates in the recent push:
* Addressed the review comments
* Added second commit to remove `/try-zulip`
* One change in the [first commit](https://github.com/zulip/zulip/pull/34237#discussion_r2023658168) -- remove `?show_try_zulip_modal` from url irrespective of the user is spectator or not.

timabbott commented on templates/corporate/case-studies/tum-case-study.md:
```diff
@@ -32,9 +32,10 @@ when Tobias came across Zulip.
-community](/try-zulip/) to see it in action. “It takes a bit of
-time to get used to, but Zulip has the best user experience of all the
-chat apps I’ve tried,” Tobias says. “With the discussion organized by
+community](https://chat.zulip.org/?show_try_zulip_modal)
```
@alya does this link want to be here, or to the main development community page?

alya replied:
Not sure. Probably https://zulip.com/development-community/ is fine here.

prakhar1144 replied:
Added a small commit (first commit) for this change.

prakhar1144 commented on docs/production/install.md:
```diff
@@ -14,8 +14,8 @@ To **import data** from [Slack][slack-import], [Mattermost][mattermost-import],
 You can **try out Zulip** before setting up your own server by [checking
-it out](https://zulip.com/try-zulip/) in the Zulip development community, or
-[creating a free test organization](https://zulip.com/new/) on Zulip Cloud.
+it out](https://chat.zulip.org/?show_try_zulip_modal) in the Zulip development community,
```
Added `URLRedirect("/try-zulip/", "https://chat.zulip.org/?show_try_zulip_modal"),` to `LANDING_PAGE_REDIRECTS`

prakhar1144 commented:
Addressed the review comments in the recent push.

timabbott commented:
Looks great, merged, thanks @prakhar1144! Please make sure to test it end-to-end after the next chat.zulip.org makes this available there.
